.github/workflows/provenance.yaml

```diff
@@ -133,3 +137,33 @@ jobs:
           OAK_FUNCTIONS_BASE_PATH: ${{ env.OAK_FUNCTIONS_BASE_PATH }}
           OAK_FUNCTIONS_UNSAFE_PATH: ${{ env.OAK_FUNCTIONS_UNSAFE_PATH }}
           OAK_BAREMETAL_CROSVM_PATH: ${{ env.OAK_BAREMETAL_CROSVM_PATH }}
+
+      # Experimental step using SLSA generic provenance generator
+      # See https://github.com/slsa-framework/slsa-github-generator/blob/main/internal/builders/generic/README.md
+      - name: Generate oak_functions_loader_base hash
+        id: hash
+        run: |
+          # first cd to the release path. This allows us calling sha256 on oak_functions_loader_base
+          # instead of having to provide the full path. This is fine for this experimental version,
+          # but has to be cleaned-up once we switch completely to SLSA provenance generation.
+          cd ${{ env.OAK_FUNCTIONS_RELEASE_PATH }}
+          # sha256sum generates sha256 hash for oak_functions_loader_base.
+          # base64 --wrap=0 disables line wrapping, and encodes to base64 and outputs on a single line.
+          echo "::set-output name=hashes::$(sha256sum oak_functions_loader_base | base64 --wrap=0)"
+
+  # This job calls the generic workflow to generate provenance.
+  # See https://github.com/slsa-framework/slsa-github-generator/blob/main/internal/builders/generic/README.md
+  provenance:
+    if: github.event_name != 'pull_request'
+    needs: [build_provenance]
+    permissions:
+      actions: read
+      id-token: write
+      contents: write
+    uses: slsa-framework/slsa-github-generator/.github/workflows/generator_generic_slsa3.yml@v1.2.0
+    with:
+      base64-subjects: '${{ needs.build_provenance.outputs.hashes }}'
+      # Set a custom name for the provenance attestation.
+      attestation-name: 'oak_functions_loader_base.intoto.jsonl'
+      # Upload provenance to a new release
```

Comment:
Out of curiousity: Where does this upload to?

Reply:
It is GitHub's internal storage or something. These files get attached to the action. Similar to the SARIF file in this action: https://github.com/project-oak/oak/actions/runs/2920567892.